functests/utils/k8sreporter/reporter.go

```diff
@@ -89,52 +90,69 @@ func (r *KubernetesReporter) SpecDidComplete(specSummary *types.SpecSummary) {
 	if !specSummary.HasFailureState() {
 		return
 	}
-	fmt.Fprintln(r.dumpOutput, "Starting dump for failed spec", specSummary.ComponentTexts)
-	r.Dump(specSummary.RunTime)
-	fmt.Fprintln(r.dumpOutput, "Finished dump for failed spec")
+	f, err := logFileFor(r.reportPath, "all", "")
+	if err != nil {
+		return
+	}
+	defer f.Close()
+	fmt.Fprintln(f, "Starting dump for failed spec", specSummary.ComponentTexts)
+	dirName := sanitize.BaseName(strings.Join(specSummary.ComponentTexts, ""))
+	dirName = strings.Replace(dirName, "Top-Level", "", 1)
+	r.Dump(specSummary.RunTime, dirName)
+	fmt.Fprintln(f, "Finished dump for failed spec")
 }
 
 // Dump dumps the relevant crs + pod logs
-func (r *KubernetesReporter) Dump(duration time.Duration) {
+func (r *KubernetesReporter) Dump(duration time.Duration, dirName string) {
```

Comment:
Can we get rid of dirName here and just use r.reportPath as our destination folder?

Reply:
r.reportPath is the main directory for the logs, dirName is per failed test
https://github.com/openshift-kni/cnf-features-deploy/blob/b139fc9480361826fd0398322e8706b87d2cf937/functests/utils/k8sreporter/reporter.go#L108
I updated New(...) to use reportPath instead of dirName

Reply:
I got it, thanks for explaining